internal/core/thirdparty/tantivy/tantivy-binding/src/index_writer.rs

```diff
@@ -0,0 +1,118 @@
+
+
+
+use futures::executor::block_on;
+
+
+
+
+use tantivy::schema::{Field, IndexRecordOption, Schema, TextFieldIndexing, TextOptions, INDEXED};
+use tantivy::{doc, tokenizer, Index, IndexWriter};
+
+use crate::data_type::TantivyDataType;
+use crate::index_reader::IndexReaderWrapper;
+
+
+pub struct IndexWriterWrapper {
+    pub field_name: String,
+    pub field: Field,
+    pub data_type: TantivyDataType,
+    pub path: String,
+    pub index: Index,
+    pub index_writer: IndexWriter,
+}
+
+impl IndexWriterWrapper {
+    pub fn create_reader(&self) -> IndexReaderWrapper {
+        IndexReaderWrapper::new(&self.index, &self.field_name, self.field)
+    }
+
+    pub fn new(field_name: String, data_type: TantivyDataType, path: String) -> IndexWriterWrapper {
+        let field: Field;
+        let mut schema_builder = Schema::builder();
+        let mut use_raw_tokenizer = false;
+        match data_type {
+            TantivyDataType::I64 => {
+                field = schema_builder.add_i64_field(&field_name, INDEXED);
+            }
+            TantivyDataType::F64 => {
+                field = schema_builder.add_f64_field(&field_name, INDEXED);
+            }
+            TantivyDataType::Bool => {
+                field = schema_builder.add_bool_field(&field_name, INDEXED);
+            }
+            TantivyDataType::Keyword => {
+                let text_field_indexing = TextFieldIndexing::default()
+                    .set_tokenizer("raw_tokenizer")
+                    .set_index_option(IndexRecordOption::Basic);
+                let text_options = TextOptions::default().set_indexing_options(text_field_indexing);
+                field = schema_builder.add_text_field(&field_name, text_options);
+                use_raw_tokenizer = true;
+            }
+            _ => {
```

Comment:
remove U64 Text in enum definition to avoid use _ to match all.

Reply:
OK.